server/api/compliance.go

```diff
@@ -0,0 +1,282 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/focalboard/server/model"
+	mmModel "github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+const (
+	complianceDefaultPage    = "0"
+	complianceDefaultPerPage = "60"
+)
+
+func (a *API) registerComplianceRoutes(r *mux.Router) {
+	// Compliance APIs
+	r.HandleFunc("/admin/boards", a.sessionRequired(a.handleGetAllBoards)).Methods("GET")
+	r.HandleFunc("/admin/boards_history", a.sessionRequired(a.handleGetBoardsHistory)).Methods("GET")
+	r.HandleFunc("/admin/blocks_history", a.sessionRequired(a.handleGetBlocksHistory)).Methods("GET")
+}
+
+func (a *API) handleGetAllBoards(w http.ResponseWriter, r *http.Request) {
+	// TODO(@pinjasaur): swagger
+
+	query := r.URL.Query()
+	teamID := query.Get("team_id")
+	strPage := query.Get("page")
+	strPerPage := query.Get("per_page")
+
+	// Valid authorization (`manage_system`)?
+	userID := getUserID(r)
+	if !a.permissions.HasPermissionTo(userID, mmModel.PermissionManageSystem) {
+		a.errorResponse(w, r, model.NewErrUnauthorized("access denied Compliance Export getAllBoards"))
+		return
+	}
+
+	// Valid license feature (Compliance)?
+	license := a.app.GetLicense()
+	if license == nil || !(*license.Features.Compliance) {
+		a.errorResponse(w, r, model.NewErrNotImplemented("insufficient license Compliance Export getAllBoards"))
+		return
+	}
+
+	if strPage == "" {
+		strPage = complianceDefaultPage
+	}
+	if strPerPage == "" {
+		strPerPage = complianceDefaultPerPage
+	}
+	page, err := strconv.Atoi(strPage)
+	if err != nil {
+		message := fmt.Sprintf("invalid `page` parameter: %s", err)
+		a.errorResponse(w, r, model.NewErrBadRequest(message))
+		return
+	}
+	perPage, err := strconv.Atoi(strPerPage)
+	if err != nil {
+		message := fmt.Sprintf("invalid `per_page` parameter: %s", err)
+		a.errorResponse(w, r, model.NewErrBadRequest(message))
+		return
+	}
+
+	boards, err := a.app.GetBoardsForTeam(teamID, page, perPage)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	// N+1; is there more?
+	// TODO: potentially fragile if len(boards) == 0 or perPage < 0
```

Comment:
better way to check `hasNext`?

Reply:
to expand on this, the "not N+1" approach would be e.g. if `per_page` is 60 then assume if the db returns 60 rows there is more data. this works except for the case of there being exactly 60 rows of data which would result in `hasNext` being `true` but the client making the final API call and expecting data but not receiving any